### DISCUSSION TITLE
Smart Contract Lottery Section

### DISCUSSION THREAD

**Discussion by AlimGlck** [Cyfrin/foundry-full-course-cu#3111] [General]
Later I am going to provide the error logs on other discussion but right now I just wanted ask did you be able to achieve to deploy the contract and make it work by only using codes. Because in video at some point he had added consumer and fund it manually on website. I tried make it work without using website and only by using code but I always got error on transferAndCall() method. So did anyone handle that just by using codes?

**Comment by EngrPips** [comment]
The only issue I know is that you will always run into an error if you try to create a subscription programmatically because of `Foundry Simulation` but if you have created a subscription on the UI and have it in your HelperConfig, then you should be able to do every other thing programmatically.
